docs/upgrading_to_v4.0.md

```diff
@@ -0,0 +1,15 @@
+# Upgrading to v4.0
+The v4.0 release of *kms* is a backwards incompatible release.
+
+### Autokey Submodule
+The current users of Autokey submodules needs to
+- Switch `project_id` to `key_project_id` (For steps check [here](#to-migrate-from-project_id-to-key_project_id))
+- Stop using `autokey_handles` field to generate keyhandles, instead directly use `google_kms_key_handle` terraform resource to create keyhandles. For detailed example check [bucket_setup_using_autokey](../examples/bucket_setup_using_autokey/).
+
+
+### To Migrate from `project_id` to `key_project_id`
```

Comment:
I don't believe we need any migration for the change from project_id to key_project_id other than changing the config which should be no-op?

```diff
- project_id = "PROJECT_ID"
+ key_project_id = "PROJECT_ID"
```